src/runtime-rs/crates/resource/src/share_fs/sandbox_bind_mounts.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2023 Alibaba Cloud
+// Copyright (c) 2023 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Note:
+// sandbox_bind_mounts supports kinds of mount patterns, for example:
+// (1) "/path/to", with default readonly mode.
+// (2) "/path/to:ro", same as (1).
+// (3) "/path/to:rw", with readwrite mode.
+//
+// sandbox_bind_mounts: ["/path/to", "/path/to:rw", "/mnt/to:ro"]
+//
+
+use std::{
+    collections::HashMap,
+    fs,
+    path::{Path, PathBuf},
+};
+
+use anyhow::{anyhow, Context, Result};
+
+use super::utils::{do_get_host_path, mkdir_with_permissions};
+use kata_sys_util::{fs::get_base_name, mount};
+
+const SANDBOX_BIND_MOUNTS_DIR: &str = "sandbox-mounts";
+const SANDBOX_BIND_MOUNTS_RO: &str = ":ro";
+const SANDBOX_BIND_MOUNTS_RW: &str = ":rw";
+
+#[derive(Clone, Default, Debug)]
+pub struct SandboxBindMounts {
+    sid: String,
+    host_mounts_path: PathBuf,
+    sandbox_bindmounts: Vec<String>,
+}
+
+impl SandboxBindMounts {
+    pub async fn new(sid: String, sandbox_bindmounts: Vec<String>) -> Result<Self> {
+        // /run/kata-containers/shared/sandboxes/<sid>/rw/passthrough/sandbox-mounts
+        let bindmounts_path =
+            do_get_host_path(SANDBOX_BIND_MOUNTS_DIR, sid.as_str(), "", true, false);
+        let host_mounts_path = PathBuf::from(bindmounts_path);
+
+        Ok(SandboxBindMounts {
+            sid,
+            host_mounts_path,
+            sandbox_bindmounts,
+        })
+    }
+
+    pub fn bind_mounts_empty(&self) -> bool {
+        if self.sandbox_bindmounts.is_empty() {
+            info!(sl!(), "sandbox bind mounts empty, skip it.");
+            return true;
+        }
+
+        false
+    }
+
+    fn parse_sandbox_bind_mounts<'a>(&self, bindmnt_src: &'a str) -> Result<(&'a str, &'a str)> {
+        // get the bindmount's r/w mode
+        let bindmount_rw = if bindmnt_src.ends_with(SANDBOX_BIND_MOUNTS_RW) {
```

Comment:
I am not sure using `if` here is ok. For example, if the input is ':is' here, IMO, reporting an error is better. Or you should examine the configuration when loading configuration.